xla/service/gpu/cudnn_norm_rewriter.cc

```diff
@@ -970,13 +970,6 @@ class CudnnNormRewriterVisitor : public DfsHloRewriteVisitor {
         VLOG(1) << "Layer norm input dimensions not supported.";
         return absl::OkStatus();
       }
-      for (int i = 0; i < norm_dims.size(); ++i) {
```

Comment:
Just to understand, is the claim that this is redundant because we check `broadcast_bias` below?

And in that case, I assume this is already covered by an existing test, right? (In which case, could you please link to it?)

Sorry, I'm not too familiar with that code, so I appreciate your help in making sure all of this is correct :)

Reply:
Yes, we verify the shape of the broadcasted scale [here](https://github.com/openxla/xla/blob/4c64418323220b2b6cd17c46e1ec15f59ecd74ac/xla/service/gpu/cudnn_norm_rewriter.cc#L982), and we separately also check the dimensions of the broadcast [here](https://github.com/openxla/xla/blob/4c64418323220b2b6cd17c46e1ec15f59ecd74ac/xla/service/gpu/cudnn_norm_rewriter.cc#L988).

Unit test [LayerNorm4D3IncorrectScaleBroadcast](https://github.com/openxla/xla/blob/4c64418323220b2b6cd17c46e1ec15f59ecd74ac/xla/service/gpu/cudnn_norm_rewriter_test.cc#L548) verifies that the norm is not rewritten when the scale has been broadcasted incorrectly.